lib/phoenix_ecto/html.ex

```diff
@@ -231,17 +231,25 @@ if Code.ensure_loaded?(Phoenix.HTML) do
     defp form_for_errors(%{action: nil}), do: []
     defp form_for_errors(%{errors: errors}), do: errors
 
-    defp form_for_hidden(data) do
-      # Since they are primary keys, we should ignore nil values.
-      for {k, v} <- Ecto.primary_key(data), v != nil, do: {k, v}
+    defp form_for_hidden(%{__struct__: module} = data) do
+      if Keyword.has_key?(module.__info__(:functions), :__schema__) do
```

Comment:
Well, I'm not sure. Perhaps there can be some legitimate uses of plain structs too? (I personally needed raw maps but plain structs support didn't seem to be so costly to add, so I did it too)